### DISCUSSION TITLE
TileSet Collision Issues

### DISCUSSION THREAD

**Discussion by Budder645** [Techokami/SonicWorldsNext#140] [Q&A] [answered]
So this has been the first engine I've actually have been able to use with my (VERY LIMITED) game design/coding skills.

So Im making a new tileset with my own sprites I've thrown together, and so far its been going great, feeling acomplished. But I then got to the tunnel and loop portion of the sprites (I made the sprites over the sprites that were in the file of the engine for reference) and I'm fairly sure I followed the tile set to a T, but for some odd reason Sonic not only clips through the tunnel like a quarter way through, but he also just clips right through the "wall" of the loop when going up as if there wasn't anything there. So Im at a complete loss here.

I REALLY want this engine to work out for me, I've put alot of effort into this since I've wanted to make my own fan game for years and everything has been working with my self teaching along the way, so I need some guidance from people who actually know what they are doing compared to me. I've put some images and videos down below in hopes to further explain my problems
![Capture 2](https://github.com/Techokami/SonicWorldsNext/assets/159749816/9bf62199-ea78-4708-854e-b490b1d135b8)
![Capture](https://github.com/Techokami/SonicWorldsNext/assets/159749816/258f3f60-4edc-4731-8d75-a5ceacbbc264)
![Capture 4](https://github.com/Techokami/SonicWorldsNext/assets/159749816/fcb78599-c104-45f5-afe8-2107337f2b47)
![Capture 3](https://github.com/Techokami/SonicWorldsNext/assets/159749816/56b419f7-908c-4ff8-a12b-8a9afb445e40)

https://github.com/Techokami/SonicWorldsNext/assets/159749816/5355b5d8-fe8f-42e3-b7ce-b2a79f93ce58

**Comment by Renhoex** [comment, marked as answer]
I think I know what happened, from the looks of it it looks like you have the collisions to only be top side enabled.

check your tile map tileset and make sure the first 4 boxes of the collision layer are enabled, if you're confused what these layers are click the three dots on the right and you'll get a clear label for what each layer is.
![image](https://github.com/Techokami/SonicWorldsNext/assets/26295900/5883333b-9035-4fbd-b419-ee74e942d28a)

let us know if this fixes the problem, and I'm glad to see for the most part you've been able to follow along with the set up so far.

**Reply by Budder645** [reply]
Just did it and it worked! Thank you so much!
